File: app/src/main/kotlin/com/easyhome/jrconsumer/app/HttpResultException.java
package com.easyhome.jrconsumer.app;
import com.easyhome.jrconsumer.mvp.extension.ErrorInfo;

import java.io.IOException;


public class HttpResultException extends IOException {
    private ErrorInfo errorInfo;

    public HttpResultException(ErrorInfo errorInfo) {
        this.errorInfo = errorInfo;
    }

    public HttpResultException(String detailMessage) {
        super(detailMessage);
    }


    public ErrorInfo getErrorInfo() {
        return errorInfo;
    }

    public void setErrorInfo(ErrorInfo errorInfo) {
        this.errorInfo = errorInfo;
    }
}

File: app/src/main/kotlin/com/easyhome/jrconsumer/mvp/extension/HttpResultException.kt
package com.easyhome.jrconsumer.mvp.extension

import java.io.IOException

class HttpResultException : IOException {
    var errorInfo: ErrorInfo? = null

    constructor(errorInfo: ErrorInfo?) {
        this.errorInfo = errorInfo
    }

    constructor(detailMessage: String?) : super(detailMessage) {}
}